### PR TITLE
Restructure API to remove confusion about 'catalog'

### LINKED ISSUE #1799: Rename or remove the `api/catalog` directory
## Problem

The majority of code for the Openverse API lives in a directory called https://github.com/WordPress/openverse/tree/main/api/catalog. Given that the Openverse catalog now lives in this repository at https://github.com/WordPress/openverse/tree/main/catalog, this name is quite confusing. 

## Description

This folder should be renamed to something else, or removed, if possible. Could all it's contents be lifted up one level into https://github.com/WordPress/openverse/tree/main/api/?

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #1799 by @zackkrida 

## Description

This PR restructure the API to remove the `catalog/` name from the codebase and reduce confusion with the top-level `catalog/` directory.

As per this change, the `api/catalog/` directory has been renamed to `api/conf/` (since it's mostly concerned configuration for the Django project) and the `api/catalog/api/` directory has been lifted up as `api/api/` (the name 'api' was retained so that all existing migrations, table_names etc. continue to remain valid).

## Testing Instructions

CI passes so that means no functionality has been impacted. If you know of any functionality in the API that is not covered by tests, please test that out, and file an issue so that we can add test coverage for it (see  #1657).

Please update the settings module in the `.env` file to `conf.settings` instead of `catalog.settings`.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by obulat — COMMENTED** [review]
I removed all the volumes and ran `just recreate`, `just init`. The API docs work fine, but I can't see any documents in the ES indices (I used Elasticvue for checking). What could be the problem?
I've tried deleting the indices again, and re-running `just init`, and still got 0 documents in image/audio/filtered-image/filtered-audio indices. The qa index had 4 or 12 items...

Could you also add a note to the testing instructions to update the `DJANGO_SETTINGS_MODULE` in .env file?

**Comment by dhruvkb** [comment]
@obulat I'll check. In the meantime, could you rebuild the image (`just dc build web`), delete the volumes (`just down -v`) and try again (`just api/init`)?

**Comment by dhruvkb** [comment]
@obulat I didn't experience any problems locally (and CI is also passing) so I'm inclined to think it's something specific to your setup. I destroyed all services and volumes, rebuilt all the images and initialised the API from scratch.

```
$ just down -v
$ just dc build
$ just api/init
```

Elasticvue shows the right number of documents.

elasticvue | Indices

**Comment by obulat** [comment]
> @obulat I didn't experience any problems locally (and CI is also passing) so I'm inclined to think it's something specific to your setup. I destroyed all services and volumes, rebuilt all the images and initialised the API from scratch.
> 
> ```
> $ just down -v
> $ just dc build
> $ just api/init
> ```
> 
> Elasticvue shows the right number of documents.
> 
> elasticvue | Indices
> 

I'll try it tomorrow. Have you reused your old .env file? After testing this PR, something broke in my setup and now I can't run main, too. I tried recreating .env fixes, but it didn't help. More testing tomorrow

**Comment by dhruvkb** [comment]
Yes @obulat, I am reusing my old `.env` file with the only change being the name of the settings module.

**Comment by obulat** [comment]
> Yes @obulat, I am reusing my old `.env` file with the only change being the name of the settings module.

Could you also try running up and init using the .env file created by running `just env` command?

**Comment by dhruvkb** [comment]
@obulat I tried deleting all the `.env` files, and recreated them with `just env`. Everything still works the same, without any issues.

File: api/api/docs/image_docs.py [hunk 1]
```diff
@@ -59,7 +59,7 @@
     eg=[image_search_list_curl],
     external_docs={
         "description": "Openverse Syntax Guide",
-        "url": "https://search.creativecommons.org/search-help",
+        "url": "https://api.creativecommons.org/search-help",
```
Comment (obulat) [inline api/api/docs/image_docs.py:62]:
```suggestion
        "url": "https://openverse.org/search-help",
```

**Review by obulat — CHANGES_REQUESTED** [review]
I re-ran the API using the commands in your previous comment, @dhruvkb, and everything worked well. Really nice improvements in the settings, and clearer folder structure.

The only blocker from me is the URL to search-help.

**Comment by obulat** [comment]
> This PR has migrations. Please rebase it before merging to ensure that conflicting migrations are not introduced.

This PR won't actually run migrations, will it?

**Review by obulat — APPROVED** [review]

**Comment by dhruvkb** [comment]
@obulat no it won't. It's saying there are migrations because the `migrations/` folder has moved.

**Review by AetherUnbound — APPROVED** [review]
LGTM, works locally, tests run locally, should be set here! Thanks for taking the time to simplify this! I love the separation of the settings too.

**Comment by sarayourfriend** [comment]
@dhruvkb This is a significant change:

> Please update the settings module in the .env file to conf.settings instead of catalog.settings.

Can you create a Make post announcing this? I just spent a good deal of time scratching my head trying to figure out where the settings had gone (there is no longer a settings.py and the new location is not obvious, I had to look at this PR to find it) and the only error you receive when running the app without making the .env file change is one about ALLOWED_HOSTS not being set. It's very confusing and I was only lucky to know that this PR existed to have a hunch of where to look to find out how to fix it.

**Pull request merged**